File: java/src/main/java/mock/auction/entity/Auction.java
package mock.auction.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

import java.sql.Date;
import java.time.LocalDateTime;
import java.util.List;

import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Auction")
public class Auction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "auction_id")
    private Integer auctionId;
    @Column(name = "start_date")
    private LocalDateTime startDate;
    @Column(name = "end_date")
    private LocalDateTime endDate;
    private String conductor;
    @Column(name = "starting_price")
    private Double startingPrice;
    @Column(name = "min_price_increase")
    private Double minPriceIncrease;
    private String period;
    @Column(name = "highest_price")
    private Double highestPrice;
    @Column(name = "auction_status")
    private String auctionStatus;
    @Column(name = "payment_status")
    private String paymentStatus;
    @Column(name = "payment_date")
    private Date paymentDate;
    @Column(name = "payment_amount")
    private Double paymentAmount;
    @Column(name = "payment_method")
    private String paymentMethod;
    @Column(name = "del_flag")
    private Boolean delFlag;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "asset_id")
    private Asset asset;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "auction_type_id")
    private AuctionType auctionType;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "winner_id")
    private AccountEntity winner;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "auction_event_id")
    private AuctionEvent auctionEvent;
}

File: php/registerparticipate/app/Models/Auction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Auction extends Model
{
    protected $table = 'auction'; // Tên bảng chính xác trong cơ sở dữ liệu

    public $timestamps = false;
    protected $primaryKey = 'auction_id';
    protected $fillable = [
        'start_date', 'end_date', 'conductor', 'asset_id', 'starting_price', 
        'min_price_increase', 'period', 'auction_type_id', 'winner_id', 'highest_price',
        'auction_status', 'auction_event_id', 'payment_status', 'payment_date', 
        'payment_amount', 'payment_method', 'del_flag'
    ];

    public function participants()
    {
        return $this->belongsToMany(Account::class, 'registparticipateauction', 'auction_id', 'account_id')
                    ->withPivot('payment', 'amount');
    }
}
